tools/manifest/sourcefile.py

```diff
@@ -218,13 +215,13 @@ def __init__(self, tests_root, rel_path, url_base, hash=None, contents=None):
 
         meta_flags = name.split(".")[1:]
 
-        self.tests_root = tests_root  # type: Union[bytes, Text]
-        self.rel_path = rel_path  # type: Union[bytes, Text]
-        self.dir_path = dir_path  # type: Union[bytes, Text]
-        self.filename = filename  # type: Union[bytes, Text]
-        self.name = name  # type: Union[bytes, Text]
-        self.ext = ext  # type: Union[bytes, Text]
-        self.type_flag = type_flag  # type: Optional[Union[bytes, Text]]
+        self.tests_root = ensure_text(tests_root)  # type: Text
```

Comment:
Nit / purely aesthetic: I might prefer treating this the same way as `rel_path` (i.e. have `tests_root_str` and `tests_root`).